app/scripts/controllers/newOverview.js

```diff
@@ -0,0 +1,1200 @@
+  // widths.
+  overview.state.breakpoint = getBreakpoint();
+  var onResize = _.throttle(function() {
+    var breakpoint = getBreakpoint();
+    if (overview.state.breakpoint !== breakpoint) {
+      $scope.$evalAsync(function() {
+        overview.state.breakpoint = breakpoint;
+      });
+    }
+  }, 50);
+
+  $(window).on('resize', onResize);
+
+  overview.renderOptions = {
+    showGetStarted: false,
+    showLoading: true
+  };
+
+  overview.filterByOptions = [{
+    id: 'name',
+    label: 'Name'
+  }, {
+    id: 'label',
+    label: 'Label'
+  }];
+  overview.filterBy = 'name';
+
+  overview.viewByOptions = [{
+    id: 'app',
+    label: 'Application'
+  }, {
+    id: 'resource',
+    label: 'Resource Type'
+  }, {
+    id: 'pipeline',
+    label: 'Pipeline'
+  }];
+
+  var getName = function(apiObject) {
+    return _.get(apiObject, 'metadata.name');
+  };
+
+  var getUID = function(apiObject) {
+    return _.get(apiObject, 'metadata.uid');
+  };
+
+  // The size of all visible top-level items.
+  var size = function() {
+    return _.size(overview.deploymentConfigs) +
+           _.size(overview.vanillaReplicationControllers) +
+           _.size(overview.deployments) +
+           _.size(overview.vanillaReplicaSets) +
+           _.size(overview.statefulSets) +
+           _.size(overview.monopods);
+  };
+
+  // The size of all visible top-level items after filtering.
+  var filteredSize = function() {
+    return _.size(overview.filteredDeploymentConfigs) +
+           _.size(overview.filteredReplicationControllers) +
+           _.size(overview.filteredDeployments) +
+           _.size(overview.filteredReplicaSets) +
+           _.size(overview.filteredStatefulSets) +
+           _.size(overview.filteredMonopods);
+  };
+
+  // Show the "Get Started" message if the project is empty.
+  var updateShowGetStarted = function() {
+    overview.size = size();
+    overview.filteredSize = filteredSize();
+
+    // Check if there is any data visible in the overview.
+    var projectEmpty = overview.size === 0;
+
+    // Check if we've loaded the top-level items we show on the overview.
+    var loaded = overview.deploymentConfigs &&
+                 overview.replicationControllers &&
+                 overview.deployments &&
+                 overview.replicaSets &&
+                 overview.statefulSets &&
+                 overview.pods;
+
+    state.expandAll = loaded && overview.size === 1;
+
+    overview.renderOptions.showGetStarted = loaded && projectEmpty;
+    overview.renderOptions.showLoading = !loaded && projectEmpty;
+
+    overview.everythingFiltered = !projectEmpty && !overview.filteredSize;
+  };
+
+  // Group a collection of resources by app label. Returns a map where the key
+  // is the app label value and the value is an array of object, sorted by
+  // `metadata.name`.
+  var groupByApp = function(collection) {
+    return AppsService.groupByApp(collection, 'metadata.name');
```

Comment:
> Or was the thinking that if we kept it as group by app but changed how the grouping worked that it would just be under this same service.

Yeah, that was the thought, although I'm OK moving it to LabelsService as well.